spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/rewrite/ModifyResponseBodyGatewayFilterFactory.java

```diff
@@ -185,8 +195,12 @@ ServerHttpResponse decorate(ServerWebExchange exchange) {
 
 					// TODO: flux or mono
 					Mono modifiedBody = clientResponse.bodyToMono(inClass)
-							.flatMap(originalBody -> config.rewriteFunction
-									.apply(exchange, originalBody));
+							.flatMap(originalBody -> config.getRewriteFunction()
```

Comment:
This isn't at all what we talked about.

Reply:
as far as I understand you asked to increase readability via change :
```
Mono modifiedBody = clientResponse.bodyToMono(inClass)
.flatMap(originalBody -> config.getRewriteFunction().apply(exchange, originalBody));

if(config.getEmptyBodyRewriteFunction() != null ){
modifiedBody = modifiedBody .switchIfEmpty(Mono.defer(() ->config.getEmptyBodyRewriteFunction());
}
//with
Mono modifiedBody = clientResponse.bodyToMono(inClass)
.flatMap(originalBody -> config.getRewriteFunction().apply(exchange, originalBody));

modifiedBody = modifiedBody.switchIfEmpty(Mono.defer(() ->
        config.getEmptyBodyRewriteFunction() != null 
            ? config.getEmptyBodyRewriteFunction() 
           : Mono.empty())
    );
```

so I thought made it even more functional and do the same but in 1 chain
```
Mono modifiedBody = clientResponse.bodyToMono(inClass)
                                      .flatMap(originalBody -> config.getRewriteFunction().apply(exchange, originalBody))
                                      .switchIfEmpty(Mono.defer(() ->
                                              config.getEmptyBodyRewriteFunction() != null 
                                              ? config.getEmptyBodyRewriteFunction() 
                                              : Mono.empty())
    );
```
Am I miss your original point or just do the bad formatting and that's why confuse you?

Reply:
@spencergibb it seems that I should fix something in this PR to make it finished.
But I miss what should be changed.
Could you please let me know if something should be changed here?